CONTRIBUTING.md

```diff
@@ -1,9 +1,197 @@
+after pushing to your branch (see [here](https://intersect-training.org/collaborative-git/03-pr/index.html) for a tutorial).
+
+Your pull request should include the following:
+- A summary including information on what you changed and why
+- References to relevant issues or discussions
+- Special notice to any portion of your changes where you have lingering questions (e.g., "was this the right way to implement this?") or want reviewers to pay special attention to
+
+Next, we will be notified of the pull request and will read it over. We will try to give an initial response quickly, and then do a longer in-depth review, at which point 
+you will probably need to respond to our comments, making changes as appropriate. We'll then respond again, and proceed in an iterative fashion until everyone is happy with the proposed 
+changes. 
+
+Once your changes are integrated, you will be added as a GitHub contributor and as one of the authors of the package. Thank you for being part of `nemos`!
+
+### Style Guide
+
+The next section will talk about the style of your code and specific requirements for certain feature development in `nemos`. 
+
+- Longer, descriptive names are preferred (e.g., x is not an appropriate name for a variable), especially for anything user-facing, such as methods, attributes, or arguments.
+- Any public method or function must have a complete type-annotated docstring (see below for details). Hidden ones do not need to have complete docstrings, but they probably should.
+
+### Testing 
+
+To run all tests, run `pytest` from within the main `nemos` repository. This may take a while as there are many tests, broken into several categories. 
+There are several options for how to run a subset of tests:
+- Run tests from one file: `pytest tests/test_glm.py`
+- Run a specific test within a specific module: `pytests tests/test_glm.py::test_func`
+- Another example specifying a test method via the command line: `pytest tests/test_glm.py::GLMClass::test_func`
+
+#### Adding tests
+
+New tests can be added in any of the existing `tests/test_*.py` scripts. Tests should be functions, contained within classes. The class contains a bunch of related tests
+(e.g., regularizers, bases), and each test should ideally be a unit test, only testing one thing. The classes should be named `TestSomething`, while test functions should be named 
+`test_something` in snakecase.
+
+If you're adding a substantial bunch of tests that are separate from the existing ones, you can create a new test script. Its name must begin with `test_`, 
+it must have an `.py` extension, and it must be contained within the `tests` directory. Assuming you do that, our github actions will automatically find it and 
+add it to the tests-to-run.
+
+> **_NOTE:_** If you have many variants on a test you wish to run, you should make use of pytest's `parameterize` mark. See the official
+> documentation [here](https://docs.pytest.org/en/stable/how-to/parametrize.html).
```

Comment:
I'd add something like, ", and see `some_test_function` for a simple example of how we use them".